qiskit/circuit/library/generalized_gates/unitary.py

```diff
@@ -111,9 +111,7 @@ def __eq__(self, other):
             return False
         if self.label != other.label:
             return False
-        # Should we match unitaries as equal if they are equal
-        # up to global phase?
-        return matrix_equal(self.params[0], other.params[0], ignore_phase=True)
+        return matrix_equal(self.params[0], other.params[0])
```

Comment:
For reference - this comment was added in #2173 at the same time as the initial `UnitaryGate` implementation, and (iirc) before the transpiler had a hard commitment to maintain global phase (or maybe more correctly, before the bugs in phase tracking were worked out).